src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -420,6 +422,28 @@ internal string GetBackupFilePath(WorkspaceModel workspace)
             return Path.Combine(BackupDirectory, fileName);
         }
 
+        /// <summary>
+        /// Backup the XML file.
+        /// </summary>
+        /// <param name="xmlDoc">The XML document.</param>
+        /// <param name="filePath">The file path.</param>
+        /// <returns></returns>
+        internal bool BackupXMLFile(XmlDocument xmlDoc, string filePath)
+        {
+            try
+            {
+                var fileName = Path.GetFileNameWithoutExtension(filePath) + "_xml";
+                var extension = Path.GetExtension(filePath);
+                var savePath = Path.Combine(this.BackupDirectory, fileName + extension);
+                xmlDoc.Save(savePath);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                return false;
```

Comment:
throw the error, but open the file even if it is not saved in the backup folder.